messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/GistSdk.kt

```diff
@@ -107,16 +113,53 @@ internal class GistSdk(
         }
 
         inAppMessagingManager.subscribeToAttribute({ it.pollInterval }) { interval ->
+            // Only manage polling when app is foregrounded
+            if (!isAppForegrounded) {
+                return@subscribeToAttribute
+            }
+
             val currentState = state
-            if (currentState.sseEnabled) {
+            if (currentState.shouldUseSse) {
                 return@subscribeToAttribute
             }
             fetchInAppMessages(duration = interval, initialDelay = interval)
         }
 
-        inAppMessagingManager.subscribeToAttribute({ it.sseEnabled }) { sseEnabled ->
-            if (sseEnabled) {
+        // Subscribe to SSE flag changes
+        inAppMessagingManager.subscribeToAttribute({ it.sseEnabled }) { _ ->
+            // Only manage polling when app is foregrounded
+            if (!isAppForegrounded) {
+                return@subscribeToAttribute
+            }
+
+            val currentState = state
+            if (currentState.shouldUseSse) {
+                // SSE is now active - stop polling
+                logger.debug("SSE enabled for identified user, stopping polling")
+                resetTimer()
+            } else {
+                // SSE not active - start polling
+                logger.debug("SSE not active, starting polling")
+                fetchInAppMessages(state.pollInterval)
```

Comment:
For cases where SSE is not enabled, wouldn't `fetchInAppMessages(state.pollInterval)` be called two times around same time? One from subscriber above for `pollInterval` and one from here?